core: Implement Aggressive Battery [1/2]

dinosnore1 opened the pull request:
Aggressive Idle is a feature that makes the device idle constants
more aggressive and ignores device motion when entering doze mode.

Aggressive Standby is a feature that stops apps from running in
the background as much.

Change-Id: Ic8da90d988f90f642e49e411d02e883f779f497b
Signed-off-by: Joe Maples <joe@frap129.org>
Signed-off-by: DennySPB <dennyspb@gmail.com>

Commit:
frameworks/base: Support for third party NFC features and extensions

This change is a combination of following changes:

1) frameworks/base: Support for third party NFC features

   Integration of below modifications are necessary to support
   third party NFC software:

   * a new interface in INfcAdapter.aidl allowing vendor specific
     extensions and features
   * a new size for MIFARE Classic tags
   * a modified constructor to distinguish MIFARE Classic tags
     from NfcA tags
   * allowing extensions to AidGroup and changing the protection
     of the instance variables to package protected

   Author: Puneet Mishra <puneetm@codeaurora.org>
   Change-Id: Ic11dc68c4ea83262c705ec50b75b5808aa064f82
   (integrated from commit 57a001b7851c97d41f042dda643f9a87aa6306e5)

2) NFC: Allow extensions to ApduServiceInfo

   Mobile network operators expect additional
   features in ApduServiceInfo to support their
   payment solutions. To achieve this, the final
   keyword is removed from the class definition
   and its instance variables, thereby allowing
   extensions to the class.

   Author: Puneet Mishra <puneetm@codeaurora.org>
   Change-Id: I9058d08e729027b04ee94405c66ff9d99c0f963b

Change-Id: I04bbe46bf9efd032e973bba7a16dc430fb6edd12
Signed-off-by: Josh Fox (XlxFoXxlX) <joshfox87@gmail.com>

AgentFabulous committed:
base: Introduce smart clock [1/2]

- Devices with notches have limited space in the statusbar area. This
  means, having a clock reduces the number of notification icons one can
  see at a time.
- Smart clock displays the clock every minute for 5 seconds.

Change-Id: I979211cc208c009b28b54a4644877ee3c507892f
Signed-off-by: Kshitij Gupta <kshitijgm@gmail.com>

Commit:
core: Implement Aggressive Battery [1/2]

Aggressive Idle is a feature that makes the device idle constants
more aggressive and ignores device motion when entering doze mode.

Aggressive Standby is a feature that stops apps from running in
the background as much.

Change-Id: Ic8da90d988f90f642e49e411d02e883f779f497b
Signed-off-by: Joe Maples <joe@frap129.org>

AgentFabulous committed:
CollpasedStatusBarFragment: Fix null object reference in smart clock

--------- beginning of crash
02-11 23:38:04.181  1711  1711 E AndroidRuntime: FATAL EXCEPTION: main
02-11 23:38:04.181  1711  1711 E AndroidRuntime: Process: com.android.systemui, PID: 1711
02-11 23:38:04.181  1711  1711 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.ContentResolver android.content.Context.getContentResolver()' on a null object reference
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at com.android.systemui.statusbar.phone.CollapsedStatusBarFragment.updateSmartClockStatus(CollapsedStatusBarFragment.java:393)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at com.android.systemui.statusbar.phone.CollapsedStatusBarFragment.access$100(CollapsedStatusBarFragment.java:58)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at com.android.systemui.statusbar.phone.CollapsedStatusBarFragment$SettingsObserver.onChange(CollapsedStatusBarFragment.java:412)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at android.database.ContentObserver.onChange(ContentObserver.java:145)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at android.database.ContentObserver$NotificationRunnable.run(ContentObserver.java:216)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:873)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:193)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6718)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:495)
02-11 23:38:04.181  1711  1711 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

Signed-off-by: Kshitij Gupta <kshitijgm@gmail.com>
Change-Id: I8e7963edff14fe5a16704d5d12a6bc51eda34074

phhusson committed:
Initial support for OnePlus in-display fingerprint sensor

Tested on OnePlus 6T.

Here are the list of things that are device-specific:
- Display modes
- fingerprint control (tell the fingerprint sensor the green circle is
ready, and what is its brightness)

Omni : Adapted for OnePlus
TODO: read circle position and size from overlays

Original commit : https://github.com/phhusson/platform_frameworks_base/commit/64b2e6d1fc378ac13783f5002e5a9d9379e75f3b

Change-Id: Ia4e0afeb3a1fa8c53d2838ece989aa9c5badd84f

Ovenoboyo committed:
Lockscreen Clock Styles: Introduce Q style text clock

A gist of what happened:
- Use a custom TextView to achieve the text
- Use Rounded Elegance font cuz why not
- Use a attribute to differenciate between hour hand and minute hand
- Use a retarded yet simple method to get Strings for given integer
- Set text according to system time format (12h/24h)
- Set a top margin to push the lockscreen date (Unuglify)
- Set a left padding for clock layout (Might need to be overlayed for some devices)

Change-Id: Icd93b98f2c6c56c618dfa3a98a3a2a8601cab902
Signed-off-by: Josh Fox (XlxFoXxlX) <joshfox87@gmail.com>

shagbag913 committed:
Aggressive Battery: fix auto enable on battery saver

Currently the receiver used for battery saver is in SmartPixelsReceiver,
however the receiver in SmartPixelsReceiver is only registered if Smart
Pixels is enabled. This commit just moves the settings changes when
battery saver is enabled to the receiver in DeviceIdleController, since
I think it makes more sense to put it in there than in
SmartPixelsReceiver.

Change-Id: I9773e246ed980170b4c582301df059f1e58f9a4f
Signed-off-by: shagbag913 <sh4gbag913@gmail.com>

Ovenoboyo committed:
CustomTextClock: Improve UI and optimisations

- Kill the massive switch loop cause I was high when I wrote it.
- Make the clock look somewhat alike to android Q's.
- Let the clock font be in accordance with the system font.
- And ofcourse add a dead overwatch meme at 12:00PM

Change-Id: I6891a2d6f717127f73ed9d5e4979fb6974a0b45c
Signed-off-by: Josh Fox (XlxFoXxlX) <joshfox87@gmail.com>

darkobas committed:
fod: differentiate dreaming and pulse

add pulse callback, change icons accordingly

Change-Id: I76758379059b846c187a7efe11835382872074e0

darkobas committed:
fod: prevent reapplying modes if theres no changes

Change-Id: Iea51d43ec1e083fb2ea1925f2ad2575a84be6670

ishubhamsingh committed:
LS clock style: Place Q style clock above keyguard status view

Like all other clock styles place this above the status view(date, alarm, weather) as well

Change-Id: Ieb922106f8338f1b491060611cc6bea717b6e20f
Signed-off-by: Josh Fox (XlxFoXxlX) <joshfox87@gmail.com>

simrat39 committed:
fod: Shishufy the fingerprint icon Because branding matters the most.

Signed-off-by: malinathani <alinathani786@gmail.com>

darkobas committed:
FOD: dont show FP icons if FOD is in use

Change-Id: I9cc5abc5ac2a2591e72aad8267f34e65ef417783

malinathani committed:
fod: Remove unnecessary defines for custom fp icon We dont need this as of now.

Signed-off-by: malinathani <alinathani786@gmail.com>

suphon-t committed:
[1/2] base: Add Q style to battery icon chooser

Change-Id: Ia321ac7a078eb49b1f3967d59df81618d1672e83

msfjarvis committed:
CustomTextClock: Code style cleanups

Also get rid of the "It's high noon" nonsense.

Change-Id: Ie40cedf5ed81012ad0a4f087167d9ec3eabc9218
Signed-off-by: Harsh Shandilya <msfjarvis@gmail.com>

msfjarvis committed:
SystemUI: Cleanup keyguard_status_view layout

Also resolve RTL problems and slightly increase font size

Change-Id: Id7d29d08393a7dd8ad6a2c873eceaded56ba6ceb
Signed-off-by: Harsh Shandilya <msfjarvis@gmail.com>

Commit:
Introduce lockscreen date styles [1/2]

Bootleggers edit: Let's use our method to set typeface per views to apply the style per view (like bold or regular), also update it the same way as LS date are refreshed and adding padding to our clocks to avoid collapsing

simrat39 commented:
@dinosnore1 Dis not gud - relaying message for shag because @shagbag913 is big dumb

static-var committed:
Q Style Clock: Set "It's" to accent color instead of gold / yellow color

aosip edit- keep exposure intact

Signed-off-by: Shreyansh Lodha <slodha96@gmail.com>
Change-Id: I769c2e2bb7b6ff047d5a7230bb70669450b42d39
Signed-off-by: Shreyansh Lodha <slodha96@gmail.com>
Signed-off-by: Josh Fox (XlxFoXxlX) <joshfox87@gmail.com>

static-var committed:
Q Style Clock: Reduce top margin, users want to see more notifications

Signed-off-by: Shreyansh Lodha <slodha96@gmail.com>
Change-Id: Idd790aaf966aae3812401285547841d2f5d7d7f8
Signed-off-by: Shreyansh Lodha <slodha96@gmail.com>

Vachounet committed:
fod: custom fingerprint icon [1/2]

eldainosor committed:
base: add our styles back [1/2]

eldainosor committed:
base: Fix the background on the small clock

Let's clean the background on small clock

eldainosor committed:
base: Let's remove custom font applying on custom date

It might be useful for those who doesn't support custom lockscreen fonts, but it ain't it chief in our case

suphon-t committed:
base: Add seekbar to media notification

Change-Id: Ic6b1b32040c4f60fdcecba6583c4f59d42c1491b

suphon-t committed:
Q Style Clock: Make it fully translatable

Change-Id: Ied91fc8f364d4fc8ed897a3c801e3b8576e11925

eldainosor committed:
base: Q seekbar: Add extra tweaks

this is to avoid the "feature" that is the seekbar overlapping on the cover art and also making the seekbar more transparent for non-seekable notifs

eldainosor committed:
base: Fixes for Monochrome clock style

The ideal clock for this might be by having the a line skip between clock and date and also the background being more solid, so this fixes all of that

eldainosor committed:
test: reduce padding on seekbar

eldainosor committed:
a small tweaks for clock date padding

eldainosor committed:
[1/2] base: add Q beta pill background

This adds this style for the date too, respecting everything from 7d5178bf37e3002304c01ea76bcf6886ec3c36a3

eldainosor committed:
Add new designs with G R A D I E N T E S

static-var committed:
KeyguardSliceView: Align as per the Clock Style

Bootleggers (@eldainosor) edit: Use our row container instead, added on ls date styles commit.

Signed-off-by: Shreyansh Lodha <slodha96@gmail.com>
Change-Id: Ie96e909cbf7c296450d8a7e2c1db115fa52484de
Signed-off-by: Harsh Shandilya <msfjarvis@gmail.com>

eldainosor committed:
Q Clock: Apply the same fonts as clock ones

eldainosor committed:
Q style: Add the padding changes for the owner text too

Like it's being done on LS slice

eldainosor committed:
q clock: cleanup

flawnn commented:
nice one actually
